kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/management/WebSocketKieServerMgmtControllerImpl.java

```diff
@@ -51,13 +51,15 @@
     @PostConstruct
     public void configure() {
         LOGGER.info("Kie Server Controller Management WebSocket service initialized");
-        if(templateStorage.isUnsatisfied()){
-            LOGGER.warn("Unable to find template storage implementation, using in memory");
+
+        // @PostConstruct can be called in Tomcat so could be null
+        if(templateStorage == null || templateStorage.isUnsatisfied()){
+            LOGGER.warn("Unable to find template storage implementation, using in memory or loaded by ServiceLoader");
         } else {
             commandService.setTemplateStorage(templateStorage.get());
         }
-        if(notificationService.isUnsatisfied()){
-            LOGGER.warn("Unable to find notification service implementation, using logging only");
+        if(notificationService == null || notificationService.isUnsatisfied()){
+            LOGGER.warn("Unable to find notification service implementation, using logging only or loaded by ServiceLoader");
         } else {
             commandService.setNotificationService(notificationService.get());
         }
```

Comment:
Additonal findings.

> why only the combination of Tomcat 9.0.50 + Java 8 calls `@PostConstruct`

Indeed, Tomcat 9.0.50 tries to find `@PostConstruct` method in `WebSocketKieServerMgmtControllerImpl` using `DefaultInstanceManager`.

https://github.com/apache/tomcat/blob/main/java/org/apache/catalina/core/DefaultInstanceManager.java#L173-L175

However, with Java 11, it fails to find the annotation because of classloader difference. (One `PostConstruct` class is loaded from `<TOMCAT_HOME>/lib/annotations-api.jar`, the other is loaded from `<KIE_CONTROLLER_APP>/WEB-INF/lib/jakarta.annotation-api-1.3.5.jar`). With Java 8, the annotation is found because `PostConstruct` class is constantly loaded from bootstrap classloader. So resulted in the issue.

In Tomcat 9.0.43 (JWS 5.5), `DefaultServerEndpointConfigurator` used for `WebSocketKieServerMgmtControllerImpl` instantiation instead of `DefaultInstanceManager` so there was no `@PostConstruct` call.

I don't see clear explanation in Servlet/WebSocket spec that Tomcat may/can call `@PostConstruct` for the websocket endpoint class. But I think we should prepare the case that `@PostConstruct` is called so this fix seems to be valid.